test/unit/testExecutorCurrentName.js

```diff
@@ -0,0 +1,150 @@
+                                {fullName: 'Cher', hasOtherName: true}
+                            ]
+                        }
+                    },
+                },
+                params: ['*']
+            };
+            const ExecutorCurrentName = steps.ExecutorCurrentName;
+            const ctx = ExecutorCurrentName.getContextData(req);
+
+            expect(ctx.index).to.equal(1);
+            done();
+        });
+
+        it('returns -1 when there is a * url param and no execs with other name', (done) => {
+            const req = {
+                session: {
+                    form: {
+                        executors: {
+                            list: []
+                        }
+                    },
+                },
+                params: ['*']
+            };
+            const ExecutorCurrentName = steps.ExecutorCurrentName;
+            const ctx = ExecutorCurrentName.getContextData(req);
+
+            expect(ctx.index).to.equal(-1);
+            done();
+        });
+    });
+
+    describe('nextStepOptions()', () => {
+
+        it('ctx.continue returns true when next executor exists in list ', (done) => {
+
+            const testCtx = {
+                index: 1,
+                list: [
+                    {lastName: 'Huggins', firstName: 'Jake', isApplying: true, isApplicant: true},
+                    {fullName: 'Madonna', hasOtherName: true},
+                    {fullName: 'Prince', hasOtherName: false},
+                    {fullName: 'Cher', hasOtherName: true}
+                ]
+            };
+
+            const ExecutorCurrentName = steps.ExecutorCurrentName;
+            const nextStepOptions = ExecutorCurrentName.nextStepOptions(testCtx);
+
+            assert.equal(testCtx.continue, true);
+
+            expect(nextStepOptions).to.deep.equal({
+                options: [
+                    {key: 'continue', value: true, choice: 'continue'},
+                ],
+            });
+            done();
+        });
+
+        it('ctx.continue returns false when at the end of exec list ', (done) => {
+
+            const testCtx = {
+                index: -1,
+                list: [
+                    {lastName: 'Huggins', firstName: 'Jake', isApplying: true, isApplicant: true},
+                    {fullName: 'Madonna', hasOtherName: true},
+                    {fullName: 'Prince', hasOtherName: false},
+                    {fullName: 'Cher', hasOtherName: true}
+                ]
+            };
+
+            const ExecutorCurrentName = steps.ExecutorCurrentName;
+            const nextStepOptions = ExecutorCurrentName.nextStepOptions(testCtx);
+
+            assert.equal(testCtx.continue, false);
+            expect(nextStepOptions).to.deep.equal({
+                options: [
+                    {key: 'continue', value: true, choice: 'continue'},
+                ],
+            });
+            done();
+        });
+
+    });
+
+    describe('action()', () => {
```

Comment:
Instead of having 2 tests here for `action()`, we could have 1 test which sets all the params to be removed and checks that they've been removed from the returned `ctx` value.

Params to be checked are:

```
ctx.index;
ctx.currentName;
ctx.continue;
ctx.isToggleEnabled;
```